### PR TITLE
Add deprecation warning for v3 images

### PR DESCRIPTION
**Pull request by zahalzel**
Work item: https://msazure.visualstudio.com/Antares/_workitems/edit/25774564/

Releasing new Host versions.
[V3 Release](PASTE_V3_LINK_HERE)
[V2 Release](PASTE_V2_LINK_HERE)

---

Adds the following deprecation message to the startup script for v3 images:

```
**WARNING**: You are using an outdated version of the Azure Functions runtime:
https://learn.microsoft.com/en-us/azure/azure-functions/migrate-version-3-version-4
====================================================================================
```

image

### PR information

- [ x] The title of the PR is clear and informative.
- [ x] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/development-docs/cleaning-up-commits.md).
- [x ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ x] New Unit tests were added for the changes made and CI is passing.

### CODE REVIEW

File: host/3.0/buster/amd64/dotnet/dotnet-isolated/start.sh
Comment (pragnagopa) [inline host/3.0/buster/amd64/dotnet/dotnet-isolated/start.sh]: any reason for not reusing /azure-functions-host/start_deprecated.sh here?

Comment (zahalzel) [reply host/3.0/buster/amd64/dotnet/dotnet-isolated/start.sh]: Made some changes to see if we can call the start_deprecated.sh script from the start.sh script in order to consolidate the warning message :)

**Review by pragnagopa — DISMISSED** [review]
Wait for approval from @Cooperlink as well.

**Review by CooperLink — APPROVED** [review]
lgtm

**Comment by arroyc** [comment]
why do we need the start_deprecate.sh in each of the language folder? it looks like same script to me? May be we can keep only single copy of it?

File: host/3.0/buster/amd64/dotnet/dotnet-inproc/dotnet-slim.Dockerfile
Comment (arroyc) [inline host/3.0/buster/amd64/dotnet/dotnet-inproc/dotnet-slim.Dockerfile:63]: are we overriding this command later stages? what is the reason of using CMD here instead of RUN?

File: host/3.0/buster/amd64/dotnet/dotnet-inproc/start_deprecated.sh
Comment (arroyc) [inline host/3.0/buster/amd64/dotnet/dotnet-inproc/start_deprecated.sh:4]: may be we can keep only this much in a new script, also i think once we are done with running the script in the build phase, we should remove it as well

File: host/3.0/buster/amd64/dotnet/dotnet-inproc/dotnet-slim.Dockerfile
Comment (CooperLink) [reply host/3.0/buster/amd64/dotnet/dotnet-inproc/dotnet-slim.Dockerfile:63]: CMD and RUN work differently. https://stackoverflow.com/questions/37461868/difference-between-run-and-cmd-in-a-dockerfile

File: host/3.0/buster/amd64/dotnet/dotnet-inproc/start_deprecated.sh
Comment (CooperLink) [reply host/3.0/buster/amd64/dotnet/dotnet-inproc/start_deprecated.sh:4]: CMD is basically the entrypoint of the script. It runs when the container is started and therefore we have to start the host at the end of any startup script

**Comment by CooperLink** [comment]
> why do we need the start_deprecate.sh in each of the language folder? it looks like same script to me? May be we can keep only single copy of it?

Single copy of this file would require us to refactor the build pipelines to handle the changed context. Out of scope for v3 images totally.

File: host/3.0/buster/amd64/dotnet/dotnet-inproc/dotnet-slim.Dockerfile
Comment (arroyc) [reply host/3.0/buster/amd64/dotnet/dotnet-inproc/dotnet-slim.Dockerfile:63]: my bad .. i was thinking this is only needed during build phase .. ignore this comment

File: host/3.0/buster/amd64/dotnet/dotnet-inproc/start_deprecated.sh
Comment (arroyc) [reply host/3.0/buster/amd64/dotnet/dotnet-inproc/start_deprecated.sh:4]: yes i understood this is sort of entrypoint for the container, ignore this

**Pull request merged**
